File: backend/backend.Domain/src/Entities/Shipment.cs
using System.Text.Json.Serialization;

namespace backend.Domain.src.Entities
{
    public class Shipment : BaseEntity
    {
        public string CompanyShipmentName { get; set; }
        public string ShipmentTrackingNumnber { get; set; }
        public ShipmentState ShipmentState { get; set; }
        public Guid OrderProductId { get; set; }

        public OrderProduct OrderProduct { get; set; }
    }

    [JsonConverter(typeof(JsonStringEnumConverter))]
     public enum ShipmentState
    {
        Delivering,
        Success,
        Cancel
    }
}

File: frontend/src/types/Shipment.ts
import { Guid } from "guid-typescript";

export interface ShipmentCreate
{
    companyShipmentName: string
    shipmentTrackingNumber: string
    shipmentState: "Delivering"|"Success"|"Cancel"
    orderProductId: Guid
}

export interface ShipmentRead
{
    id: Guid
    companyShipmentName: string
    shipmentTrackingNumber: string
    shipmentState: "Delivering" | "Success" | "Cancel"
    orderProductId: Guid
}

export interface OrderProductShipmentRead
{
    companyShipmentName: string
    shipmentTrackingNumber: string
    shipmentState: "Delivering" | "Success" | "Cancel"
}
